R/Synthetic_data_gen#6.R

```diff
@@ -0,0 +1,166 @@
+  concept_table <- generate_concept_table()
+  
+  # Generate synthetic drug_exposure table with increased records
+  drug_exposure <- data.frame(
+    drug_exposure_id = 1:num_records,
+    person_id = sample(1:20, num_records, replace = TRUE),  # 20 unique persons
+    drug_concept_id = sample(1:20, num_records, replace = TRUE),  # Randomly choose from 20 concepts
+    drug_exposure_start_date = sample(seq(as.Date("2023-01-01"), as.Date("2023-10-01"), by="day"), num_records, replace = TRUE),
+    quantity = sample(1:5, num_records, replace = TRUE),  # Random quantity between 1 and 5
+    dose_unit_source_value = dose_units[sample(1:20, num_records, replace = TRUE)],  # Assign directly from dose_units based on concept
+    sig = sample(c("Take one tablet daily", "Take two tablets twice a day", 
+                   "Take three tablets every six hours", "As needed"), num_records, replace = TRUE)  
+  )
+  
+  # Merge to include concept names
+  drug_exposure <- merge(drug_exposure, concept_table, by.x = "drug_concept_id", by.y = "concept_id", all.x = TRUE)
+  
+  # Assign route_concept_id using the assign_route_id function
+  drug_exposure$route_concept_id <- sapply(drug_exposure$concept_name, assign_route_id)
+  
+  # Patterns to standardize the "sig" field
+  patterns <- c(
+    "daily" = "once a day|daily|1 time a day",
+    "twice daily" = "twice a day|2 times a day|two tablets twice a day",
+    "three times daily" = "three times a day|3 times a day|three tablets every six hours",
+    "four times daily" = "four times a day|4 times a day",
+    "every other day" = "every other day|every second day",
+    "weekly" = "weekly|once a week",
+    "as needed" = "as needed|prn|if necessary"
+  )
+  
+  # Function to apply pattern matching and replace sig values
+  standardize_sig <- function(sig, patterns) {
+    for (standardized in names(patterns)) {
+      # Apply pattern replacement based on regex matches
+      sig <- gsub(patterns[standardized], standardized, sig, ignore.case = TRUE)
+    }
+    return(sig)
+  }
+  
+  # Apply the function to the 'sig' column in drug_exposure
+  drug_exposure <- drug_exposure %>%
+    mutate(sig = standardize_sig(sig, patterns))
+  
+  # Ensure end dates are always after start dates
+  drug_exposure <- drug_exposure %>%
+    mutate(drug_exposure_end_date = drug_exposure_start_date + sample(1:30, num_records, replace = TRUE))  # Add random days between 1 to 30 to the start date
+  
+  # Rearrange columns to ensure drug_exposure_end_date comes after drug_exposure_start_date
+  drug_exposure <- drug_exposure %>%
+    select(drug_exposure_id, person_id, drug_concept_id, drug_exposure_start_date, drug_exposure_end_date, 
+           quantity, dose_unit_source_value, route_concept_id, sig)
+  
+  return(drug_exposure)
+}
+
+
+#Generate concept data
+concept_table <- generate_concept_table()
+
+print(concept_table)
+
+# Generate drug exposure data
+drug_exposure_data <- generate_drug_exposure(100)

```

Comment:
- When I ran this part of the script I got this error message -  Error in drug_exposure %>% mutate(sig = standardize_sig(sig, patterns)) : could not find function "%>%"
- I then loaded the library(dplyr) and re-ran the script. It worked and produced the concept and drug exposure tables. 
- At the start of the script the command library(dplyr) should be included

Reply:
Thanks! I imported the library at the beginning of the script.